next/Dockerfile

```diff
@@ -36,14 +23,29 @@ LABEL org.opencontainers.image.revision="${GIT_COMMIT}" \
       org.opencontainers.image.licenses="EUPL-1.2"
 ENTRYPOINT [ "/sbin/tini", "--" ]
 
+FROM build-base AS install-prod
+ENV NPM_CONFIG_CACHE=/build/.npm
+RUN --mount=type=cache,target=/build/.npm \
+    npm config set cache /build/.npm \
+ && npm ci --omit=dev --frozen-lockfile
+
+FROM install-prod AS build
+COPY . ./
+ENV NEXT_TELEMETRY_DISABLED=1
+RUN --mount=type=cache,target=/build/.npm \
+    npm config set cache /build/.npm \
+    && npm ci --include=dev --frozen-lockfile \
+    && npm run build
+
```

Comment:
is the reordering that happened here important ? if so please comment why this needs to happen after the previous command. If not please revert.